### PR TITLE
feat(crosswalk): ignore occlusions in the presence of traffic lights

### PR DESCRIPTION
**Pull request by maxime-clem**
## Description

Required launch PR: https://github.com/autowarefoundation/autoware_launch/pull/919

## Tests performed

Psim

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp [hunk 1]
```diff
@@ -241,7 +241,8 @@ bool CrosswalkModule::modifyPathVelocity(PathWithLaneId * path, StopReason * sto
     return t && cmp_fn((now - *t).seconds(), planner_param_.occlusion_time_buffer);
   };
   const auto is_crosswalk_ignored =
-    (planner_param_.occlusion_ignore_with_red_traffic_light && isRedSignalForPedestrians()) ||
+    (planner_param_.occlusion_ignore_with_traffic_light &&
+     !road_.regulatoryElementsAs<const lanelet::TrafficLight>().empty()) ||
```
Comment (dkoldaev) [inline planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp:245]: maybe create an inline `_roadHasTrafficLight()` for this to make more readable?

Comment (dkoldaev) [reply planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp:245]: also looks like `isRedSignalForPedestrians` is not used anymore.

**Review by dkoldaev — COMMENTED** [review]
Looks good, but I think it's better to cleanup if there are unused functions.

File: planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp (hunk 1 above)
Comment (maxime-clem) [reply planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp]: Thank you for your review.
`isRedSignalForPedestrians` is used somewhere else so I cannot remove it.
I made an intermediate variable `crosswalk_has_traffic_light` to make the condition more readable. I also fixed a bug (we care about the crosswalk's traffic light, not the road's).
[0bebece](https://github.com/autowarefoundation/autoware.universe/pull/6604/commits/0bebece9a62eeb94e56aa6b06d63fea31514e722)

**Review by dkoldaev — APPROVED** [review]

**Review by takayuki5168 — APPROVED** [review]

**Pull request merged**